libwasmvm/src/iterator.rs

```diff
@@ -97,4 +115,60 @@ impl GoIter {
         };
         (result, gas_info)
     }
+
+    pub fn next_key(&mut self) -> BackendResult<Option<Vec<u8>>> {
+        self.next_key_or_val(self.vtable.next_key_db)
+    }
+
+    pub fn next_value(&mut self) -> BackendResult<Option<Vec<u8>>> {
+        self.next_key_or_val(self.vtable.next_value_db)
+    }
+
+    #[inline(always)]
+    fn next_key_or_val(
+        &mut self,
+        next_fn: Option<
+            extern "C" fn(
+                iterator_t,
+                *mut gas_meter_t,
+                *mut u64,
+                *mut UnmanagedVector, // output
+                *mut UnmanagedVector, // error message output
+            ) -> i32,
+        >,
+    ) -> BackendResult<Option<Vec<u8>>> {
+        let next_db = match next_fn {
+            Some(f) => f,
+            None => {
+                let result = Err(BackendError::unknown("iterator vtable not set"));
```

Comment:
Would be good to mention which field of the vtable is not set in those error messages